packaging/win/delugewin.spec

```diff
@@ -2,30 +2,33 @@
 # -*- mode: python ; coding: utf-8 -*-
 import os
 import sys
-import deluge.common
-from PyInstaller.compat import is_win
-from PyInstaller.utils.hooks import collect_all, copy_metadata
+from PyInstaller.utils.hooks import collect_submodules, copy_metadata
 
 datas = []
 binaries = []
-hiddenimports = []
+hiddenimports = ['pygame']
 
 # Collect Meta Data
 datas += copy_metadata('deluge', recursive=True)
 datas += copy_metadata('service-identity', recursive=True)
 
 # Add Deluge Hidden Imports
-tmp_ret = collect_all('deluge')
-hiddenimports += tmp_ret[2]
+hiddenimports += collect_submodules('deluge')
 
-#Add Hidden Imports for Plugins
-tmp_ret2 = collect_all('twisted')
-hiddenimports += tmp_ret2[2]
+# Add Hidden Imports for Plugins
+# Add stdlib as Hidden Imports.
+# This is filtered list that excludes some common examples or stuff not useful in plugins (such as tty, mailbox, tutledemo etc.).
+# It is safe to assume that 90% of that list would already be included anyway.
+stdlib = [ 'string', 're', 'unicodedata', 'struct', 'codecs', 'datetime', 'zoneinfo', 'calendar', 'collections', 'array', 'weakref',
+'types', 'copy', 'enum', 'numbers', 'math', 'cmath', 'decimal', 'fractions', 'random', 'statistics', 'itertools', 'functools',
+'operator', 'pathlib', 'fileinput', 'stat', 'tempfile', 'glob', 'fnmatch', 'shutil', 'pickle', 'copyreg', 'shelve', 'marshal',
+'dom', 'sqlite3', 'zlib', 'gzip', 'bz2', 'lzma', 'csv', 'hashlib', 'hmac', 'secrets', 'os', 'io', 'time', 'logging', 'platform',
+'errno', 'queue', 'socket', 'ssl', 'email', 'json', 'mimetypes', 'base64', 'binhex', 'binascii', 'quopri', 'uu', 'html', 'xml',
+'urllib', 'http', 'ftplib', 'smtplib', 'uuid', 'xmlrpc.client', 'ipaddress', 'locale', 'sys' ]
+for module in stdlib:
+    hiddenimports += collect_submodules('twisted', filter=lambda name: 'test' not in name)
```

Comment:
This is still broken - you are collecting `twisted` in a loop.

Reply:
> In addition, adding this before we collect on `twisted` should solve our worries with plugins:
> 
> ```python
> # Add stdlib as Hidden Imports.
> # This is filtered list that excludes some common examples or stuff not useful in plugins (such as tty, mailbox, tutledemo etc.).
> # It is safe to assume that 90% of that list would already be included anyway.
> stdlib = [ 'pickle', 'json', 'email', 'http', 'marshal', 'uuid', 'base64', 'string', 'struct', 'glob', 'ssl', 'urllib', 'html', 'sys', 
>   'crypt', 'datetime', 'hmac', 'locale', 'queue', 're', 'enum', 'collections', 'tokenize', 'hashlib', 'xml', 'csb', 'ipaddress', 'os' ]
> for module in stdlib:
>   hiddenimports += collect_submodules(module, filter=lambda name: 'test' not in name)
> ```
copy pasted from your comment didn't really look at this.

Reply:
my bad fixed it.